### PR TITLE
feature/cordova

### PR DESCRIPTION
**Pull request by lukeisontheroad**

### CODE REVIEW

**Comment by davidyuk** [comment]
Can we use [this configuration](https://github.com/okode/ionic-travis/blob/master/.travis.yml) to build for both Android and iOS on Travis?

**Comment by emil-apeunit** [comment]
@davidyuk We need to use jenkins:
 - We need a dedicated macos runner for the ios version, travis didn't handle this very well in the past: https://github.com/travis-ci/travis-ci/issues/7304
 - Build might become heavy and block the other travis jobs
So pls let's stay with jenkins since we have the setup anyways

File: Dockerfile.ci [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+COPY package.json package.json
+RUN npm install
+
+ADD . .
+
+# As cordova changes global modules when adding a platform, we have to build the apk already during the docker build process
+RUN npm run build
+RUN cordova platform rm android &&cordova platform add android
+RUN npm run build:android
```
[the remark this answers is not in the document]
Comment (lukeisontheroad) [reply Dockerfile.ci]: The first run is (kind of) needed as it generates the www directory. If there is no www directory, cordova will fail with the message "Error: Current working directory is not a Cordova-based project." Alternatively we can create an empty www folder.

File: config.xml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    </feature>
+    <edit-config file="*-Info.plist" mode="merge" target="NSCameraUsageDescription">
+        <string>To scan QR Codes</string>
+    </edit-config>
+    <engine name="browser" spec="^5.0.3" />
+    <engine name="ios" spec="^4.5.4" />
+    <engine name="android" spec="^6.3.0" />
+    <plugin name="cordova-plugin-whitelist" spec="^1.3.3" />
+    <plugin name="cordova-plugin-splashscreen" spec="^5.0.2" />
+    <plugin name="phonegap-plugin-barcodescanner" spec="^7.1.2">
```
[the remark this answers is not in the document]
Comment (lukeisontheroad) [reply config.xml]: I used the phonegap-plugin-barcodescanner plugin for testing, I removed it now.

**Comment by davidyuk** [comment]
Most of this PR is merged in #156

**Pull request closed without merging**